source/features/repo-age.tsx

```diff
@@ -4,50 +4,87 @@ import cache from 'webext-storage-cache';
 import select from 'select-dom';
 import repoIcon from 'octicon/repo.svg';
 import elementReady from 'element-ready';
+import fetchDom from '../libs/fetch-dom';
 import features from '../libs/features';
-import * as api from '../libs/api';
-import {getRepoGQL, getRepoURL} from '../libs/utils';
+import {getRepoURL} from '../libs/utils';
 
 const dateFormatter = new Intl.DateTimeFormat('en-US', {
 	year: 'numeric',
 	month: 'long',
 	day: 'numeric'
 });
 
-const getRepoCreationDate = cache.function(async (): Promise<string> => {
-	const {repository} = await api.v4(`
-		repository(${getRepoGQL()}) {
-			createdAt
-		}
-	`);
+type OldestCommitDetails = {
+	url: string;
+	datetime: string;
+};
 
-	return repository.createdAt;
+const getOldestCommitDetails = cache.function(async (): Promise<OldestCommitDetails | false> => {
+	await elementReady<HTMLAnchorElement>('.commit-tease-sha', {target: select('.commit-tease')!});
+	const commitsCount = Number(select('li.commits .num')!.textContent!.replace(',', ''));
+	const lastCommitHash = select<HTMLAnchorElement>('.commit-tease-sha')!.href.split('/').pop()!;
+
+	if (commitsCount === 0) {
+		return false;
+	}
+
+	if (commitsCount === 1) {
+		return {
+			url: `/${getRepoURL()}/commit/${lastCommitHash}`,
+			datetime: select('.commit-tease-sha + span relative-time')!.getAttribute('datetime')!
+		};
+	}
+
+	const oldestCommit = await fetchDom(
+		`${getRepoURL()}/commits?after=${lastCommitHash}+${commitsCount - 2}`, 'ol:last-child > li.commits-list-item'
+	);
+
+	return {
+		url: select<HTMLAnchorElement>('.commit-title [href*="/commit/"]', oldestCommit)!.href,
+		datetime: select('relative-time', oldestCommit)!.getAttribute('datetime')!
+	};
 }, {
+	isExpired: (cachedValue: OldestCommitDetails | false) => !cachedValue || typeof cachedValue.url === 'undefined',
```

Comment:
`isExpired` is only called if the age was cached, it can't be false. Does this type work?

```suggestion
 	// eslint-disable-next-line @typescript-eslint/no-explicit-any
	isExpired: (cachedValue: any) => typeof cachedValue === 'string',
```

Reply:
It will work but I was trying to avoid `any`.

Reply:
Yeah the correct type would be `string | OldestCommitDetails` (because the cache is either in the old format or the new format) but `webext-storage-cache` doesn't allow that type, I have to fix it.

Reply:
> `isExpired` is only called if the age was cached, it can't be false. Does this type work

I was going through the `webext-storage-cache` code and found that it will cache the `false` value also. We are returning `false` when no of commits are 0. Also, I found that at a couple of places we are returning false from the getter functions but in that case, the false values will be cached. Am I making sense?

```
const freshValue = await getter(...args);
await set<TValue>(key, freshValue, options.expiration);
return freshValue;
```

Reply:
So I can change the type to 
```
isExpired: (cachedValue: OldestCommitDetails | false | string) => !cachedValue || typeof cachedValue === 'string'
```
and it will work as well as reflect the correct types.